content/developers/developer-patterns/navigating-merklelogs/index.md

```diff
@@ -43,40 +36,24 @@ the dynamically sized portions of the format, please see [Massf Blob Offset Tabl
+    due originally to [Peter Todd](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2016-May/012715.html), though much parallel invention has occurred. They have been independently analysed in the context of [cryptographic asynchronous accumulators](https://eprint.iacr.org/2015/718.pdf), Generalised multi proofs for [Binary Numeral Trees](https://eprint.iacr.org/2021/038.pdf). And also by the [ethereum research community](https://ethresear.ch/t/batching-and-cyclic-partitioning-of-logs/536).
 
-Each contains the verifiable data for a fixed number, and sequential
-range, of your events. The number of events is determined by your log
-configuration parameter `massif height`. Currently all logs have a massif
-height of `14` And the number of event *leaf* log entries in each massif is
-2<sup>height-1</sup>, which is 2<sup>14-1</sup> leaves, which is `8192` leaves [^1].
+Each massif contains the verifiable data for a fixed number, and sequential range, of your events. The number of events is determined by your log configuration parameter `massif height`. Currently all logs have a massif height of `14` And the number of event *leaf* log entries in each massif is 2<sup>height-1</sup>, which is 2<sup>14-1</sup> leaves, which is `8192` leaves [^2].
 
-[^1]: From time to time, we may re-size your massifs. We are able to do this
-    without impacting the verifiability of the contained data and without
-    invalidating your previously cached copies taken from the earlier massif
-    size configuration. Simple binary file compare operations can show that the
-    verifiable data for the new configuration is identical to that in the
-    original should you wish to assure your self of this fact.
+[^2]: From time to time, we may re-size your massifs. We are able to do this without impacting the verifiability of the contained data and without invalidating your previously cached copies taken from the earlier massif size configuration. Simple binary file compare operations can show that the verifiable data for the new configuration is identical to that in the original should you wish to assure your self of this fact.
 
-Here, we have drawn `massif n` as open ended to illustrate that the last massif
-is always in the process of being *appended* to.
+Here, we have drawn `massif n` as open ended to illustrate that the last massif is always in the process of being *appended* to.
 
-[Massf Blob Offset Tables](/developers/developer-patterns/massif-blob-offset-tables) gives you a shortcut for picking the right massif. It can also be fairly easily computed from only the `merklelog_entry.commit.index` *mmrIndex* on your event using the example javascript on that page.
+[Massif Blob Pre-Calculated Offsets](/developers/developer-patterns/massif-blob-offset-tables) gives you a shortcut for picking the right massif. It can also be fairly easily computed from only the `merklelog_entry.commit.index` *mmrIndex* on your event using the example javascript on that page.
```

Comment:
```suggestion
[Massif Blob Pre-Calculated Offsets](/developers/developer-patterns/massif-blob-offset-tables) gives you a shortcut for picking the right massif.
It can also be fairly easily computed from only the `merklelog_entry.commit.index` *mmrIndex* on your event using the example javascript on that page.
```